modules/Classroom/components/ExamList/ExamList.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+
+import { Button, Collapse, Flex, SimpleGrid, Title, Loader } from "@mantine/core";
+import { showNotification } from "@mantine/notifications";
+import { FaChevronRight, FaChevronDown } from "react-icons/fa";
+
+import { useGetExamsQuery } from "@/shared/redux/rtk-apis/exams/exams.api";
+
+import ExamCard from "../ExamCard/ExamCard";
+import { filterExamsByDate } from "./ExamList.helpers";
+import { IExamsProps } from "./ExamList.interface";
+import { useStyles } from "./ExamList.styles";
+
+const ExamList: React.FC<IExamsProps> = ({ classroomId }) => {
+  const { classes } = useStyles();
+  const [toggleExamsCollapse, setToggleExamsCollapse] = useState(true);
+  const {
+    data: exams,
+    isLoading,
+    error,
+  } = useGetExamsQuery({ classroomId }, { skip: !classroomId });
+
+  if (error) {
+    showNotification({
+      title: "Error",
+      message: "Failed to load exams.",
+      color: "red",
+    });
+  }
+
+  const { pastExams, futureExams } = exams
+    ? filterExamsByDate(exams)
+    : { pastExams: [], futureExams: [] };
+
+  return (
+    <>
+      <Flex>
+        <Button
+          variant="subtle"
+          size="compact"
+          leftIcon={
+            toggleExamsCollapse ? <FaChevronDown color="white" /> : <FaChevronRight color="white" />
+          }
+          className={classes.titleButton}
+          onClick={() => setToggleExamsCollapse(!toggleExamsCollapse)}
+        >
+          <Title my={"md"} order={3} className={classes.collapseTitle}>
+            Exams
+          </Title>
+        </Button>
+      </Flex>
+
+      <Collapse in={toggleExamsCollapse} transitionDuration={0} animateOpacity={false}>
+        {isLoading ? (
+          <Loader size="sm" />
+        ) : (
+          <>
+            {futureExams && futureExams?.length > 0 && (
```

Comment:
`futureExams && futureExams?.length > 0 ` in this block the use `?.`  is redundant here since you are already checking whether `futureExams` is null or underfined.